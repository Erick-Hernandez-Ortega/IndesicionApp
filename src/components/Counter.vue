<template>
    <h2>Counter {{ title }}</h2>
    <p>{{ count }} <sup>2</sup> = {{ squareValue }}</p>

    <div>
        <button type="button" @click="count++">+1</button>
        <button type="button" @click="count--">-1</button>
    </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'; // Por TS

export default defineComponent ({
    name: "Counter",
    props: {
        title: {
            type: String,
            default: 'Counter',
            validator (value: string): boolean {
                return value.length > 5
            }
        }
    },
    data () {
        return {
            count: 0
        }
    },
    methods: {
        getSquareValue (): number {
            return this.count * this.count
        }
    },
    computed: {
        squareValue (): number {
            return this.count * this.count
        }
    }
});
</script>


<style scoped>
button {
    background-color: #64bb87;
    border-radius: 5px;
    border: 1px solid white;
    color: white;
    cursor: pointer;
    margin: 0 5px;
    padding: 5px 15px;
    transition: all 0.3s ease-in-out;
}

button:hover {
    background-color: #5aa67b;
    transition: 0.3s ease-in-out;
}
</style>