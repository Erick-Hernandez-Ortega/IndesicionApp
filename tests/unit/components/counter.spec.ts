import Counter from "@/components/Counter.vue";
import { shallowMount } from '@vue/test-utils';  // Probar componentes de vue

describe('Counter view', () => { 
    test('Debe de hacer match con el snapshot', () => { 
        // Al hacer esto por primera vez se crea un snapshot y se guarda
        const wrapper = shallowMount(Counter);
        expect(wrapper.html()).toMatchSnapshot();
     });
 })