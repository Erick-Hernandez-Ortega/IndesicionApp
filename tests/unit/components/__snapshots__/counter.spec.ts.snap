// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter view Debe de hacer match con el snapshot 1`] = `
<h2>Counter Counter</h2>
<p>0 <sup>2</sup> = 0</p>
<div><button type="button">+1</button><button type="button">-1</button></div>
`;
